docs/proposals/20210720-kdm-template.md

```diff
@@ -0,0 +1,97 @@
+
+After spending time evaluating CAPI's in-tree `KubeadmControlPlaneProvider` we think this provider can give us the base components necessary using the standardized KubeadmControlPlane spec to generate the necessary pod objects. 
+
+## Motivation
+
+- Alignment with CAPI core components
+- Enable `Kubeadm` based specs for templating the control plane
+
+### Goals
+
+- Reuse `KubeadmControlPlane`(KubeadmControlPlane) for CAPN component configurations
+
+### Non-Goals/Future Work
+
+- Implement `NestedMachine`'s this will be documented and implemented in an additional CAEP
+- Use `KubeadmControlPlane` to do health checking, this comes with the `NestedMachine`
+
+## Proposal
+
+### Templating the Control Plane
+
+In this section, we will focus on the details of how to templating the control plane and would not discuss the details of how `NestedMachine` controller works. Assume that we have installed the `KubeadmControlPlaneController`, the `NestedMachineController`, and all necessary CAPI controllers on the meta cluster. We have also imported the `kubeadm` packages from the `kubernetes` module. Then, the templating process will include the following steps.
+
+1. The user applies the `KubeadmControlPlane` CR and all other necessary CAPI CRs.
+2. The `KubeadmControlPlane` controller keeps checking if the `NestedCluster` CR is ready.
+3. If yes, the `KubeadmControlPlane` controller use the `NestedMachineTemplate` to creates desired number of `NestedMachines` for the `NestedCluster` Infrastructure.
+4. The `KubeadmControlPlane` controller generates the `KubeadmConfig` CR based on the `KubeadmControlPlane` created in the first step. 
+5. The `KubeadmControlPlane` controller assigns the `KubeadmConfig` to the owned `NestedMachine` CR by setting the ObjectReference.
+6. The `NestedMachine` controller keeps checking if the `KubeadmConfig` has been assigned to the `NestedMachine` CR.
+7. If yes, the `NestedMachine` controller calls functions from the `kubeadm` package to generate the static pod template.
+8. The `NestedMachine` controller extracts the `PodTemplateSpec` from the static pod template generated in the sixth step and creates the StatefulSet template.
+
+![Templating the control plane](images/nestedcontrolplane/kdm-template.svg)
+
+### User Stories
+
+1. As the `NestedMachine` controller, I would like to create a `NestedControlPlane` with the same templates as the kubernetes control plane created by the `kubeadm`.
```

Comment:
All the other user stories make sense except this one is a little confusing, can you clarify what you mean?